src/main/java/com/palantir/gradle/revapi/RevapiPlugin.java

```diff
@@ -46,40 +47,52 @@ public void apply(Project project) {
 
         RevapiExtension extension = project.getExtensions().create("revapi", RevapiExtension.class, project);
 
-        Configuration revapiNewApi = project.getConfigurations().create("revapiNewApi", conf -> {
-            conf.extendsFrom(project.getConfigurations().getByName(JavaPlugin.RUNTIME_ELEMENTS_CONFIGURATION_NAME));
-        });
-
         ConfigManager configManager = new ConfigManager(configFile(project));
-
-        TaskProvider<RevapiReportTask> revapiTask = project.getTasks().register("revapi", RevapiReportTask.class);
+        File resultsFile = new File(project.getBuildDir(), "revapi/revapi-results.json");
+
+        TaskProvider<RevapiAnalyzeTask> results = project.getTasks()
+                .register("revapiAnalyze", RevapiAnalyzeTask.class, task -> {
+                    Configuration revapiNewApi = project.getConfigurations().create("revapiNewApi", conf -> {
+                        conf.extendsFrom(
+                                project.getConfigurations().getByName(JavaPlugin.RUNTIME_ELEMENTS_CONFIGURATION_NAME));
+                    });
+                    Provider<OldApi> oldApi = ResolveOldApi.oldApiProvider(project, extension, configManager);
+
+                    task.dependsOn(allJarTasksIncludingDependencies(project, revapiNewApi));
+                    task.getAcceptedBreaks().set(acceptedBreaks(project, configManager, extension.oldGroupAndName()));
+
+                    Jar jarTask = project.getTasks().withType(Jar.class).getByName(JavaPlugin.JAR_TASK_NAME);
+                    task.getNewApiJars().set(jarTask.getOutputs().getFiles());
+                    task.getNewApiDependencyJars().set(revapiNewApi);
+                    task.getOldApiJars().set(oldApi.map(OldApi::jars));
+                    task.getOldApiDependencyJars().set(oldApi.map(OldApi::dependencyJars));
+
+                    task.getResultsFile().set(resultsFile);
+                });
+
+        TaskProvider<RevapiReportTask> revapiTask = project.getTasks()
+                .register("revapi", RevapiReportTask.class, task -> {
+                    task.dependsOn(results);
+                    task.getResultsFile().set(resultsFile);
```

Comment:
maybe call it analysisResultsFile

Reply:
could just put `results.get().getResultsFile()` inside the set, then `dependsOn` is no longer necessary